### ISSUE TITLE
Push notifications not being received by android device

### ISSUE THREAD

**Issue by miracle7** [ParsePlatform/parse-server#1577]
I have managed to get push notifications working in a test app using Unity, Parse Server and Heroku, however when I try the same configuration in our actual game, the push notifications never come through.

I am running Parse Server 2.2.7 on Heroku with MongoDB, and the test app does not have anything that the main app is missing.

All other Parse functionality is working fine, even the installation is saved with a device token, however when I send a push notification if the device token was generated by the test app it works fine, but if it is using the token generated by our actual game I see this in the _PushStatus document in my DB, and it never gets through to the phone:

> {
>     "_id": "uUrP5xF64W",
>     "pushTime": "2016-04-21T01:07:43.653Z",
>     "_created_at": {
>         "$date": "2016-04-21T01:07:43.653Z"
>     },
>     "query": "{}",
>     "payload": "{\"alert\":\"Please work\"}",
>     "source": "rest",
>     "status": "succeeded",
>     "numSent": 0,
>     "pushHash": "1c5dd358f9bd636feff13872764e096b",
>     "_wperm": [],
>     "_rperm": [],
>     "numFailed": 1,
>     "failedPerType": {
>         "android": 1
>     }

I can't figure out how to troubleshoot this, I need more than 'it failed for android' to try and debug this but I don't know where to look as there is nothing in the logs.

I tried going through this repo trying to figure out where 'numFailed' was incremented, and eventually dug all the way down to [PushAdapter.js](https://github.com/ParsePlatform/parse-server/blob/7afc08abe4914482e790054f78d5ae112e554222/src/Adapters/Push/PushAdapter.js) but had no luck finding any more information as to what determines whether the device token is valid or whatever is blocking it.

If I copy the device token from the test app to the install generated by the new app it all works fine as well, so it is definitely an issue with the way the device registers or the way the push notification is trying to be sent.

Would really appreciate any help on this as I am very stuck, just basically doing trial and error to see if stuff works at this point.

**Comment by miracle7** [comment]
For some context on how I got to this point, you can take a look at the issue I opened here:
https://github.com/ParsePlatform/Parse-SDK-dotNET/issues/190

In short I got push notifications working for Parse.com, and they broke when I moved to parse server, though everything else kept working.

**Comment by flovilmart** [comment]
Try running your server with VERBOSE=1 this will log the response from GCM.

**Comment by miracle7** [comment]
I started with the parse server example on heroku, do you know how I would go about enabling verbose on there?

Sorry if this is a really basic question, I am very new to a lot of this stuff.

**Comment by drew-gross** [comment]
One way is to run your Parse Server with the `verbose` option:

```js
var myServer = ParseServer({
  verbose: true,
  ...otherOptions
});
```

**Comment by miracle7** [comment]
Thank you, doing it like that didn't work, but I was able to enable it by putting this in my config variables in the heroku dashboard:
![image](https://cloud.githubusercontent.com/assets/5317556/14733216/ba40c320-08a4-11e6-90c5-b8f0f57c1ff1.png)

The response did say 'mismatched sender id' which allowed me to debug the application and find the problem:

We have another native java plugin in our app, but last time someone compiled that they accidentally bundled custom notification handler into it as well (which meant using old PARSE_GCM_SENDER_ID), which was being used instead of the new one I compiled.

In other words, my bad and thanks for your help :)
